language-plutus-core/src/Language/PlutusCore/Constant/Dynamic/BuiltinName.hs

```diff
@@ -56,3 +58,9 @@ dynamicAppend = dynamicBuiltinNameAsTerm dynamicAppendName
 
 dynamicTraceName :: DynamicBuiltinName
 dynamicTraceName = DynamicBuiltinName "trace"
+
+dynamicTraceMeaning :: DynamicBuiltinNameMeaning
+dynamicTraceMeaning = DynamicBuiltinNameMeaning sch (flip trace ()) where
+    sch =
+        Proxy @String `TypeSchemeArrow`
+        TypeSchemeResult (Proxy @())
```

Comment:
This definition was omitted intentionally. The thing is, we do not want to denote `Plutus.trace` as `Haskell.trace`, instead we want `Plutus.trace` to collect elements in a list and return that list to the caller, so that the caller can do with traced messages whatever he wants (e.g. display them in the playgrounds). Confusing, I know.

However your definition does make sense and is convenient in certain cases, so I propose to keep it, but rename it to `dynamicTraceMeaningMock` for example, so that it's clear that this is not the real meaning of `Plutus.trace`. And then also add `dynamicTraceDefinitionMock :: DynamicBuiltinNameDefinition` please (see the same module for examples).

Reply:
Will do.